plugins/ai/copilot-chat.nix

```diff
@@ -0,0 +1,316 @@
+            '';
+          };
+        };
+
+        default = {
+          Explain.prompt = "/COPILOT_EXPLAIN Write an explanation for the active selection as paragraphs of text.";
+          Review = {
+            prompt = "/COPILOT_REVIEW Review the selected code.";
+            callback = ''
+              function(response, source)
+                -- see config.lua for implementation
+              end
+            '';
+          };
+          Fix.prompt = "/COPILOT_GENERATE There is a problem in this code. Rewrite the code to show it with the bug fixed.";
+          Optimize.prompt = "/COPILOT_GENERATE Optimize the selected code to improve performance and readablilty.";
+          Docs.prompt = "/COPILOT_GENERATE Please add documentation comment for the selection.";
+          Tests.prompt = "/COPILOT_GENERATE Please generate tests for my code.";
+          FixDiagnostic = {
+            prompt = "Please assist with the following diagnostic issue in file:";
+            selection = "require('CopilotChat.select').diagnostics";
+          };
+          Commit = {
+            prompt = "Write commit message for the change with commitizen convention. Make sure the title has maximum 50 characters and message is wrapped at 72 characters. Wrap the whole message in code block with language gitcommit.";
+            selection = "require('CopilotChat.select').gitdiff";
+          };
+          CommitStaged = {
+            prompt = "Write commit message for the change with commitizen convention. Make sure the title has maximum 50 characters and message is wrapped at 72 characters. Wrap the whole message in code block with language gitcommit.";
+            selection = ''
+              function(source)
+                return select.gitdiff(source, true)
+              end
+            '';
+          };
+        };
+      in
+      helpers.defaultNullOpts.mkAttrsOf (with types; either str promptType) default ''
+        Default prompts.
+      '';
+
+    window = {
+      layout =
+        helpers.defaultNullOpts.mkEnumFirstDefault
+          [
+            "vertical"
+            "horizontal"
+            "float"
+            "replace"
+          ]
+          ''
+            Layout for the window.
+          '';
+
+      width =
+        helpers.defaultNullOpts.mkNullable (with types; either (number.between 0.0 1.0) ints.positive) 0.5
+          ''
+            Fractional width of parent, or absolute width in columns when > 1.
+          '';
+
+      height =
+        helpers.defaultNullOpts.mkNullable (with types; either (number.between 0.0 1.0) ints.positive) 0.5
+          ''
+            Fractional height of parent, or absolute height in rows when > 1.
+          '';
+
+      relative =
+        helpers.defaultNullOpts.mkEnumFirstDefault
+          [
+            "editor"
+            "win"
+            "cursor"
+            "mouse"
+          ]
+          ''
+            Relative position.
+            (Only for floating windows.)
```

Comment:
Single line breaks don't render. If you wanted an actual line break you'd need to end the line with `\`, or for a full paragraph break you need a blank line.

I think this applies to a few option descriptions.

If they're rendering as you intended then ignore this.